test/helpers/setup-helper.js

```diff
@@ -73,14 +73,15 @@ const popoverContent = window.document.createElement('div');
 popoverContent.setAttribute('id', 'popover-content');
 window.document.body.appendChild(popoverContent);
 
-// fetch
+// Fetch
 // fetch is part of node js in future versions, thus triggering no-shadow
 // eslint-disable-next-line no-shadow
-const fetch = require('node-fetch');
+const { default: fetch, Headers, Request, Response } = require('node-fetch');
 
-/* eslint-disable-next-line no-shadow */
-const { Headers, Request, Response } = fetch;
 Object.assign(window, { fetch, Headers, Request, Response });
+// some of our libraries currently assume that `fetch` is globally available,
+// so we need to assign this for tests to run
+global.fetch = fetch;
```

Comment:
With the `assets-controller` upgrade, the tests for DetectTokensController started failing. A real NetworkController gets used in these tests, and that causes the block tracker to start, which makes a call to the network through the provider. The fetch middleware uses a utility function from `controller-utils` to do this, and that ultimately uses `global.fetch` to make the request. Explicitly assigning `fetch` like this made those tests work again.

I was surprised that we haven't needed to do this before. In upgrading `assets-controllers`, `controller-utils` also got upgraded. However, `isomorphic-fetch` was removed in `controller-utils` back in 3.0.0, and here we've upgraded `controller-utils` from 3.4.0 to 4.0.0. Perhaps some other library that previously imported `isomorphic-fetch` is no longer doing so.